website/docs/integrations/atlantis.md

```diff
@@ -149,23 +161,498 @@ workflows:
+or `--ssh-key-password` flags are provided at the same time.
+
+The command works by:
+
+- Cloning the target branch (`--ref`) or checking out the commit (`--sha`) of the remote target branch, or using the already cloned target repository
+  specified by the `--repo-path` flag
+
+- Deep merging all stack configurations for both the current working branch and the remote target branch
+
+- Looking for changes in the component directories
+
+- Comparing each section of the stack configuration looking for differences
+
+- Generating the `atlantis.yaml` file with the `projects` sections consisting of a list of the affected Atlantis projects
+
+Since Atmos first checks the component folders for changes, if it finds any affected files, it will mark all related components and stacks as
+affected. Atmos will then skip evaluating those stacks for differences since we already know that they are affected.
+
+Refer to [`atmos atlantis generate repo-config`](/cli/commands/atlantis/generate-repo-config) for the description of the command and all flags.
+
+## Working with Private Repositories
+
+If the flag `--affected-only=true` is passed on the command line (e.g. `atmos atlantis generate repo-config --affected-only=true`), the command
+will clone and checkout the remote target repo (which can be the default `refs/heads<default_branch>` reference, or specified by the command-line
+flags `--ref`, `--sha` or `--repo-path`). If the remote target repo is private, special attention needs to be given to how to work with private
+repositories.
+
+There are a few ways to work with private repositories with which the current local branch is compared to detect the changed files and affected Atmos
+stacks and components:
+
+- Using the `--ssh-key` flag to specify the filesystem path to a PEM-encoded private key to clone private repos using SSH, and
+  the `--ssh-key-password` flag to provide the encryption password for the PEM-encoded private key if the key contains a password-encrypted PEM block
+
+- Execute the `atmos atlantis generate repo-config --affected-only=true --repo-path <path_to_cloned_target_repo>` command in
+  a [GitHub Action](https://docs.github.com/en/actions). For this to work, clone the remote target repository using
+  the [checkout](https://github.com/actions/checkout) GitHub action. Then use the `--repo-path` flag to specify the path to the already cloned
+  target repository with which to compare the current branch
+
+## Using with GitHub Actions
+
+If the `atmos atlantis generate repo-config --affected-only=true` command is executed in a [GitHub Action](https://docs.github.com/en/actions), and
+you don't want to store or generate a long-lived SSH private key on the server, you can do the following:
+
+- Create a GitHub
+  [Personal Access Token (PAT)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)
+  with scope permissions to clone private repos
```

Comment:
This is only required if the action is attempting to clone a repo which is not itself.